### PR TITLE
[FEQ]Yaswanth/FEQ-1440/Added Dropdown component

### PR DESCRIPTION
**Pull request by yaswanth-deriv**
Added a dropdown component within the deriv/ui package, making it accessible for use across various packages by importing it directly from this library.

### CODE REVIEW

File: lib/components/Dropdown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    dropdownIcon: React.ReactNode;
+    errorMessage?: InputProps['message'];
+    icon?: React.ReactNode;
+    isRequired?: boolean;
+    label?: InputProps['label'];
+    list: {
+        text?: React.ReactNode;
+        value?: string;
+    }[];
+    listHeight?: Extract<TGenericSizes, 'lg' | 'md' | 'sm'>;
```
Comment (shayan-deriv) [inline lib/components/Dropdown/index.tsx:21]: you don't need to export the type in Text component
```suggestion
    listHeight?: Extract<React.ComponentProps<typeof TGenericSizes>, 'lg' | 'md' | 'sm'>;
```

File: lib/components/Dropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import React, { isValidElement, useCallback, useEffect, useState } from 'react';
+import clsx from 'clsx';
+import { useCombobox } from 'downshift';
+import { TGenericSizes } from '../Text';
```
Comment (shayan-deriv) [inline lib/components/Dropdown/index.tsx:4]:
```suggestion

```

File: lib/components/Text/index.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import clsx from 'clsx';
 
 import './Text.scss';
 
-type TGenericSizes = '2xl' | '2xs' | '3xl' | '3xs' | '4xl' | '5xl' | '6xl' | 'lg' | 'md' | 'sm' | 'xl' | 'xs';
+export type TGenericSizes = '2xl' | '2xs' | '3xl' | '3xs' | '4xl' | '5xl' | '6xl' | 'lg' | 'md' | 'sm' | 'xl' | 'xs';
```
Comment (shayan-deriv) [inline lib/components/Text/index.tsx:6]:
```suggestion
type TGenericSizes = '2xl' | '2xs' | '3xl' | '3xs' | '4xl' | '5xl' | '6xl' | 'lg' | 'md' | 'sm' | 'xl' | 'xs';
```

File: lib/components/Input/index.tsx [hunk 4]
```diff
@@ -4,9 +4,10 @@ import HelperMessage from "./HelperMessage";
 import "./Input.scss";
 
 export type InputVariants = "general" | "success" | "error";
-interface InputProps
+export interface InputProps
```
Comment (shayan-deriv) [inline lib/components/Input/index.tsx:7]: you're not importing it anywhere so there is no need to export it 🙏 
```suggestion
 interface InputProps
```

File: lib/components/Dropdown/index.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    }[];
+    listHeight?: Extract<TGenericSizes, 'lg' | 'md' | 'sm'>;
+    name: InputProps['name'];
+    onChange?: (inputValue: string) => void;
+    onSelect: (value: string) => void;
+    value?: InputProps['value'];
+    variant?: 'comboBox' | 'prompt';
+};
+
+export const Dropdown: React.FC<TProps> = ({
```
Comment (aizad-deriv) [inline lib/components/Dropdown/index.tsx:29]: I don't suggest we using `React.FC` it doesn't add any value to the component and more over it also has a few underlying issues.

File: lib/components/Dropdown/index.tsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+            reactNode.forEach(function (child) {
+                string += reactNodeToString(child);
+            });
+        } else if (isValidElement(reactNode)) {
+            string += reactNodeToString(reactNode.props.children);
+        }
+        return string;
+    };
+    const { closeMenu, getInputProps, getItemProps, getMenuProps, getToggleButtonProps, isOpen, openMenu } =
+        useCombobox({
```
Comment (shayan-deriv) [inline lib/components/Dropdown/index.tsx:65]: can we move this to separate hook?

File: lib/components/Dropdown/index.tsx (hunk 5 above)
Comment (shayan-deriv) [reply lib/components/Dropdown/index.tsx]: We were using  `react.FC` previously but it  lead to issues for the consumer. Hence, we decided  to incorporate type safety in the conventional manner, as opposed to using.

**Review by shayan-deriv — APPROVED** [review]

**Pull request merged**